x-pack/plugins/enterprise_search/public/applications/app_search/components/result_settings/query_performance/query_performance.tsx

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+
+import { useValues } from 'kea';
+
+import { EuiBadge, EuiBadgeProps } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+
+import { ResultSettingsLogic } from '../result_settings_logic';
+
+enum QueryPerformanceRating {
+  Optimal = 'Optimal',
+  Good = 'Good',
+  Standard = 'Standard',
+  Delayed = 'Delayed',
+}
+
+type QueryPerformanceBadgePropsCollection = {
+  [x in QueryPerformanceRating]: EuiBadgeProps;
+};
+
+type QueryPerformanceBadgeContentCollection = {
+  [x in QueryPerformanceRating]: string;
+};
```

Comment:
just wondering - is this a good candidate for a `Record<>` instead? no worries if not

Reply:
Yes, I think so. This was carryover as well.

Reply:
cab6ee6dc9e5d2492b04478d57aee860d7145daa